parachain/pallets/ethereum-beacon-client/src/lib.rs

```diff
@@ -130,290 +114,261 @@ pub mod pallet {
 		InvalidAttestedHeaderSlot,
 		DuplicateFinalizedHeaderUpdate,
 		InvalidFinalizedPeriodUpdate,
-		InvalidExecutionHeaderUpdate,
+		ExecutionHeaderAlreadyImported,
 		FinalizedBeaconHeaderSlotsExceeded,
 		ExecutionHeaderMappingFailed,
 		BLSPreparePublicKeysFailed,
 		BLSVerificationFailed(BlsError),
 	}
 
+	/// Latest imported finalized block root
 	#[pallet::storage]
-	pub(super) type FinalizedBeaconHeaders<T: Config> =
-		StorageMap<_, Identity, H256, BeaconHeader, OptionQuery>;
+	#[pallet::getter(fn latest_finalized_block_root)]
+	pub(super) type LatestFinalizedBlockRoot<T: Config> = StorageValue<_, H256, ValueQuery>;
 
+	/// Beacon state by finalized block root
 	#[pallet::storage]
-	pub(super) type FinalizedBeaconHeaderStates<T: Config> =
-		StorageValue<_, BoundedVec<FinalizedHeaderState, T::MaxFinalizedHeadersToKeep>, ValueQuery>;
```

Comment:
`FinalizedHeaderState` used here is for pruning obselete data in `FinalizedBeaconState`. I would prefer we keep it.

Reply:
Resolved in https://github.com/Snowfork/snowbridge/pull/840

Reply:
Well, after thinking about the pruning algorithm some more, I don't think we should be using `FinalizedBeaconHeaderStates` for pruning. That was a mistake, I believe.

`FinalizedBeaconHeaderStates` was originally introduced to help the offchain relayer generate ancestry proofs for execution headers updates. The storage item only needed contain the last few dozen recently imported finalized header states.

Then for pruning, we decided to reuse `FinalizedBeaconHeaderStates` for another purpose - to determine an older finalized header state we could prune.  This was a mistake.

However this introduces a very bad constraint. If we want to keep the last months of beacon headers, then we need to store a month of state in `FinalizedBeaconHeaderStates`. But this storage item is a vector. We can't keep so much data in a vector in storage. This will eventually cause weight problems.

So I want to us to eliminate `FinalizedBeaconHeaderStates`. I don't think its necessary. I'm certain we can change the algorithm in the relayer to not rely on it.

And for pruning finalized beacon state, that's not a priority right now. We can introduce the Ring Buffer approach in another PR. But not right now - I want us to focus on these cleanups first. We can revisit pruning at a later time.

Reply:
Yes, that make sense. Just remove pruing stuff.

Reply:
> If we want to keep the last months of beacon headers
I think not need to cache that much data, based on the `calculateClosestCheckpointSlot` algorithm [here](https://github.com/Snowfork/snowbridge/blob/8f6ed317d7558eec8a14441e7dc9dd5d07bb0096/relayer/relays/beacon/cache/cache.go#L138-L156) checkpoint only valid in range of 1 period. Consider `finalized_update` changes very infrequent(i.e. almost 12mins) in testnet, that means `27*60/12=135` checkpoints at most. 

Any way, I agree we can make this PR focus on ALC spec compliant and revisit pruning at a later time.